### PR TITLE
Build with Go 1.23

### PR DESCRIPTION
**Pull request by stefanprodan**
Part of: https://github.com/fluxcd/flux2/issues/4927

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,15 +1,15 @@
-FROM alpine:3.19 as builder
+FROM alpine:3.20 as builder
 
 RUN apk add --no-cache ca-certificates curl
 
 ARG ARCH=linux/amd64
-ARG KUBECTL_VER=1.30.0
+ARG KUBECTL_VER=1.33.0
```
Comment (makkes) [inline Dockerfile:6]:
```suggestion
ARG KUBECTL_VER=1.31.0
```
?

**Review by makkes — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (stefanprodan) [reply Dockerfile]: Good catch! Thanks

**Comment by makkes** [comment]
I suppose we should also update `.github/runners/prereq.sh` to use kubectl 1.31.

**Comment by stefanprodan** [comment]
The runners have been decommissioned, we no longer use Equinix. We are now using GitHub EE ARM runners.

**Pull request merged**